src/braket/default_simulator/simulator.py

```diff
@@ -43,7 +41,7 @@ class DefaultSimulator(BraketSimulator):
     DEVICE_ID = "default"
 
     def run(
-        self, circuit_ir: Program, qubit_count: int, shots: int = 0, *, batch_size: int = 1,
+        self, circuit_ir: Program, qubit_count: int, shots: int, simulation: Simulation,
```

Comment:
I would be extremely careful about making a change like this, since it looks like a breaking change. You need to be extremely careful about making sure that existing code doesn't break.

In particular, if you make this change, does the code in the SDK still work? I.e. when I create a LocalSimulator in the SDK, does it have the same behavior as the old code?

Reply:
A better way of doing this ideally would be to keep the existing implementation of DefaultSimulator, and add another DensityMatrixSimulator, with different backend type.